mindsdb/integrations/handlers/anomaly_detection_handler/anomaly_detection_handler.py

```diff
@@ -0,0 +1,171 @@
+    model_names = [model_names] if model_names is None else model_names
+    model_names = [model_names] if type(model_names) is str else model_names
+    return model_names
+
+
+class AnomalyDetectionHandler(BaseMLEngine):
+    """Integration with the PyOD and CatBoost libraries for
+    anomaly detection. Both supervised and unsupervised.
+    """
+
+    name = "anomaly_detection"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.generative = True  # makes unsupervised learning work
+
+    def create(self, target, df, args={}):
+        """Train a model and save it to the model storage"""
+        using_args = args["using"]
+        model_type = using_args["type"] if "type" in using_args else None
+
+        model_names = get_model_names(using_args)
+
+        model_save_paths = []
+        model_targets = []
+        model_class_names = []
+        for model_name in model_names:
+            model = choose_model(df, model_name=model_name, model_type=model_type, target=target)
+            this_model_target = "outlier" if target is None else target  # output column name for unsupervised learning
+            save_path = "model.joblib" if model_name is None else model_name + ".joblib"
+            dump(model, save_path)
+            model_save_paths.append(save_path)
+            model_targets.append(this_model_target)
+            model_class_names.append(model.__class__.__name__)
+        model_args = {"model_path": model_save_paths, "target": model_targets, "model_name": model_class_names}
+        self.model_storage.json_set("model_args", model_args)
+
+    def predict(self, df, args={}):
+        """Load a model from the model storage and use it to make predictions"""
+        model_args = self.model_storage.json_get("model_args")
+        results_list = []
+        for model_path in model_args["model_path"]:
+            model = load(model_path)
+            if "__mindsdb_row_id" in df.columns:
+                df = df.drop("__mindsdb_row_id", axis=1)
+            if model_args["target"][0] in df.columns:
+                df = df.drop(model_args["target"], axis=1)
+            predict_df = preprocess_data(df).astype(float)
+            results = model.predict(predict_df)
+            results_list.append(results)
+        final_results = np.array(results_list).mean(axis=0)
+        final_results = np.where(final_results > 0.5, 1, 0)
+        return pd.DataFrame({model_args["target"][0]: final_results})
+
+    def describe(self, attribute="model"):
+        model_args = self.model_storage.json_get("model_args")
+        df = pd.DataFrame({"model_name": [], "target": []})
+        if attribute == "model":
+            for model_name, target in zip(model_args["model_name"], model_args["target"]):
+                df2 = pd.DataFrame({"model_name": model_name, "target": target}, index=[0])
+                df = pd.concat([df, df2], ignore_index=True)
+            return df
```

Comment:
Would suggest raising a simple exception for any other `attribute`s that are not supported (i.e. `!= model`).